modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,147 @@
+to stay on top of any Conda updates (if Personal Machine), or rebuilding PETSc and libMesh (if using
+[!ac](HPC) resources).
+
+### Applying Upstream Changes
+
+If this is your first time following these instructions there should be no updates. However, we will
+perform an update to familiarize yourself with the entire process.
+
+{{ApplicationName}} developers will expect users to have performed the following sections, whenever
+they says "I have updated {{ApplicationName}}".
+
+```bash
+cd ~/projects/{{binary}}
+git fetch upstream
+```
+
+`git fetch upstream` updates your local references with that of the remote. It does not actually
+modify any source files. It will return nothing if there is nothing new to update.
+
+You need to know your upstream head branch. This will be the branch you will want to continuously
+rebase against. That is, keeping your local {{binary}} repository up to date. The following command
+will reveal the upstream head branch name:
+
+```bash
+git remote show upstream | grep "HEAD branch" | sed 's/.*: //'
+```
+
+Next, we will perform a `rebase` against the provided upstream branch name:
+
+```bash
+git rebase upstream/branch   # replace 'branch' with the discovered branch name
+```
+
+With {{binary}} updated, we can update its submodules.
+
+### Updating Submodules and Conda Packages
+
+Most MOOSE-based applications require additional support libraries available in the form of
+git submodules. To update submodules, perform the following:
+
+```bash
+cd ~/projects/{{binary}}
+git submodule update --init
+```
+
+Remember the cyclical warnings earlier? By performing a submodule update, you should now perform
+either a Conda update, or rebuilding PETSc/libMesh:
+
+- If +Personal Machine (Conda)+:
+
+  ```bash
+  mamba activate moose
+  mamba update --all
+  ```
+
+- If +[!ac](INL) [!ac](HPC) Sawtooth or Lemhi+:
+
+  It will be necessary to build PETSc and libMesh using the supplied build scripts in
+  `moose/scripts`. The time it takes to build both PETSc and libMesh can be hours:
+
+  ```bash
+  module load use.moose moose-dev    # a reminder to have these loaded
+  export MOOSE_JOBS=6 METHODS=opt    # a reminder to set these
+  cd ~/projects/{{binary}}/moose
+  scripts/update_and_rebuild_petsc.sh
+  scripts/update_and_rebuild_libmesh.sh
+  ```
+
+  !alert tip title=MOOSE_JOBS? METHODS?
+  Go to [MOOSE Build System](application_development/build_system.md) for more information on
+  available methods. MOOSE_JOBS controls how many cores to use during the use of our scripts.
+
+### Running Make
+
+With everything up to date, we should clean out any previous build of {{ApplicationName}}, and build
+anew:
+
+```bash
+cd ~/projects/{{binary}}
+make clobberall
+make -j 6
+```
+
+`make clobberall` deletes the stale object/library files left behind from a previous build. While not
+always necessary, please get into the habit of performing this operation before running `make`.
```

Comment:
I think this is a little too strict. Maybe add an `!alert note` suggesting that this should be done as a troubleshooting step whenever a build fails and mention why.

Reply:
I can make this an alert instead.

What makes sense to you a few months later, when they are trying to perform an update?

This instead?
```bash
git clean -xfd
git submodule foreach --recusive git clean -xfd
```

Reply:
Added an alert annotation on running `make clobberall` if the make fails (and if they are trying to perform an update)